src/js/components/locations-configuration/EditZoneModal.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { Form } from 'react-final-form';
+import { getTranslate } from 'react-localize-redux';
+import { connect } from 'react-redux';
+import Alert from 'react-s-alert';
+
+import { hideSpinner, showSpinner } from 'actions';
+import CheckboxField from 'components/form-elements/CheckboxField';
+import TextField from 'components/form-elements/TextField';
+import apiClient from 'utils/apiClient';
+import { renderFormField } from 'utils/form-utils';
+import Translate, { translateWithDefaultMessage } from 'utils/Translate';
+
+import 'components/locations-configuration/AddZoneModal.scss';
+
+const FIELDS = {
+  active: {
+    type: CheckboxField,
+    label: 'react.locationsConfiguration.addZone.status.label',
+    defaultMessage: 'Status',
+    attributes: {
+      withLabel: true,
+      label: 'Active',
+    },
+  },
+  name: {
+    type: TextField,
+    label: 'react.locationsConfiguration.name.label',
+    defaultMessage: 'Name',
+    attributes: {
+      required: true,
+      withTooltip: true,
+      tooltip: 'react.locationsConfiguration.addZone.name.tooltip.label',
+    },
+  },
+};
+
+const validate = (values) => {
+  const requiredFields = ['name'];
+  return Object.keys(FIELDS)
+    .reduce((acc, fieldName) => {
+      if (!values[fieldName] && requiredFields.includes(fieldName)) {
+        return {
+          ...acc,
+          [fieldName]: 'react.default.error.requiredField.label',
+        };
+      }
+      return acc;
+    }, {});
+};
+
+
+class EditZoneModal extends Component {
```

Comment:
you don't need a separate component for editing, it's almost identical to the addZoneModal